requirements.txt

```diff
@@ -25,6 +25,7 @@ pandas
 portalocker
 psutil
 pyqt5==5.11.3
+pyqt5-tools
```

Comment:
> (Perhaps calling show caused some sort of strange state, given that it was called on a headless Windows machine?)

Ah, okay. [66de345](https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/3916/commits/66de3459bce921135fbdd0b4753fff4630bf7205) proves that theory wrong! Commenting out just `label.show()` still causes the error.

So, it must be one of the former lines (`QApplication`, `QLabel`)?

I can try playing with this tomorrow, perhaps on my local Windows environment, to see if a non-headless environment can provide some clues.

Reply:
Some updates:

* I get error code `-1` on my local Windows environment, too. So, it has nothing to do with running `PyQT5` on a headless machine.
* To save time in debugging, I commented out everything in `sct_check_dependencies` _except_ PyQT5... and what do you know, the PyQT5 code runs fine with exit code `0`!
* I then slowly isolate bits of `sct_check_dependencies` to identify the offending interaction. I found that:
    * If only the `matplotlib` check or the `pyqt5` check is present, the script exits with exit code `0`.
    * But, if both the `matplotlib` _and_ `pyqt5` checks are present, then the script exits with exit code `-1`.
 
For kicks, I tried swapping the order of the `matplotlib` and `pyqt5` checks, and... exit code `0`. Huh.

Reply:
For kicks, I tried swapping the order of the `matplotlib` and `pyqt5` checks, and... exit code `0`. Huh.

So, from what I can tell:

* Calling `matplotlib.pyplot.fig` loads the `QtAgg` backend. If you try to create a `pyqt5` application afterwards, then PyQT5 throws a fit. 
* Creating a `pyqt5` application _first_ (with no `matplotlib` backend loaded) works fine. If you then call `matplotlib.pyplot.fig` afterwards, there's no error.

I don't really understand the internals of what's going on here, but the lesson I'm getting is that [you shouldn't mix `pyplot` and `pyqt5`](https://stackoverflow.com/questions/57680015/extra-plot-window-and-crash-with-matplotlib-and-pyqt-on-windows#comment101807827_57680015). 

Thankfully, SCT doesn't do this in its actual scripts (everywhere we use `PyQt5`, we _don't_ use `pyplot`); this only happens in the contrived scenario caused by `sct_check_dependencies`. So, I think it should be okay to live with this hacky workaround of simply swapping the order of checks.

Reply:
This is some epic debugging, well done!

This fix is probably fine for now, but here's an idea if you want to make it more robust: maybe we could factor out the PyQt5 test into its own script, and call it via `run_proc` (or similar) so that it's isolated in its own process? Because it looks like [PyQt5 can cause crashes on exit](https://www.riverbankcomputing.com/static/Docs/PyQt5/gotchas.html#crashes-on-exit). (I'm not sure if this is really what's happening, but since PyQt5 is both a wrapper around C++ libraries and an application framework, I'm not surprised that it would interact weirdly with other code, hence the suggestion of isolating it.)

Btw, thanks for committing your printf-debugging in addition to posting the results! It's very helpful to see both.

Reply:
Also, it looks like there's a [recommended way of mixing matplotlib (hence pyplot) and Qt](https://matplotlib.org/stable/gallery/user_interfaces/embedding_in_qt_sgskip.html), which further confirms that they interact strangely.